client/src/App.tsx

```diff
@@ -19,28 +20,30 @@ function App(): JSX.Element {
         <SnackBarProvider>
           <AuthProvider>
             <SocketProvider>
-              <NavBar />
-              <Switch>
-                <Route exact path="/login" component={Login} />
-                <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/my-jobs">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/messages">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/my-sitters">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/edit-profile" component={EditMenu} />
+              <NotificationProvider>
```

Comment:
Do we want to wrap the entire app with the `NotificationProvider`? Each time we receive a notification the app will be re-rendered - impacting overall app performance :)

Reply:
My thoughts were that the provider component by default (not custom) only re-renders when they have to (i.e when part of its state changes reference). According to that article it seems we are safe as we use custom providers throughout the app; in this case we are able to simply wrap the navbar.

Reply:
@ugur-akin thanks a lot for providing the article, I tested things and it's safe to say that the above implementation of the `NotifcationProvider` should not hinder performance. My bad Trevor! 😄